src/main/java/com/mraof/minestuck/tileentity/redstone/WirelessRedstoneTransmitterTileEntity.java

```diff
@@ -42,20 +46,42 @@ public void tick()
 		tickCycle++;
 	}
 	
-	public BlockPos getDestinationBlockPos()
+	public BlockPos getDestinationBlockPosFromOffset()
 	{
-		if(destBlockPos == null)
-			destBlockPos = new BlockPos(0, 0, 0);
-		return destBlockPos;
+		if(offsetPos == null)
+			offsetPos = new BlockPos(0, 0, 0);
+		if(facing == null)
+			facing = getBlockState().getValue(WirelessRedstoneTransmitterBlock.FACING);
+		
+		Direction stateFacing = getBlockState().getValue(WirelessRedstoneTransmitterBlock.FACING);
+		
+		//in cases where the block has been rotated after the coordinates were set(rotator blocks or placement in structure templates), this will transform the relative coordinates by rotating them around the block
+		if(facing != stateFacing)
+			this.offsetPos = offsetPos.rotate(MSRotationUtil.rotationBetween(this.facing, stateFacing));
+		
+		//refreshing the facing value now that offsetPos has been corrected to fix the change
+		this.facing = getBlockState().getValue(WirelessRedstoneTransmitterBlock.FACING);
+		
+		return getBlockPos().offset(offsetPos);
 	}
 	
-	public void setDestinationBlockPos(BlockPos destinationPosIn)
+	public void setOffsetFromDestinationBlockPos(BlockPos destinationPosIn)
 	{
-		this.destBlockPos = destinationPosIn;
+		if(level != null) //level is null when gotten from Load
+		{
+			this.facing = level.getBlockState(worldPosition).getValue(WirelessRedstoneTransmitterBlock.FACING);
```

Comment:
Get the block state as an argument from the `load()` function instead.

Reply:
Not sure what you mean

Reply:
Instead of getting the block state from the world/level, get it as a function argument to `setOffsetFromDestinationBlockPos()`.